apps/dashboard/src/hooks/use-create-workflow.ts

```diff
@@ -0,0 +1,57 @@
+import { createWorkflow } from '@/api/workflows';
+import { useEnvironment } from '@/context/environment/hooks';
+import { QueryKeys } from '@/utils/query-keys';
+import { buildRoute, ROUTES } from '@/utils/routes';
+import { type CreateWorkflowDto, WorkflowCreationSourceEnum } from '@novu/shared';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useNavigate } from 'react-router-dom';
+import { z } from 'zod';
+import { workflowSchema } from '../components/workflow-editor/schema';
+
+interface UseCreateWorkflowOptions {
+  onSuccess?: () => void;
+}
+
+export function useCreateWorkflow({ onSuccess }: UseCreateWorkflowOptions = {}) {
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const { currentEnvironment } = useEnvironment();
+
+  const mutation = useMutation({
+    mutationFn: async (workflow: CreateWorkflowDto) => createWorkflow({ environment: currentEnvironment!, workflow }),
+    onSuccess: async (result) => {
+      await queryClient.invalidateQueries({ queryKey: [QueryKeys.fetchWorkflows, currentEnvironment?._id] });
+      await queryClient.invalidateQueries({
```

Comment:
This query invalidation shouldn't be necessary as we've just created a workflow with a unique workflowId. No cache collisions are expected. Am I missing something?

Reply:
It was a refactor and copy from the existing implementation, I guess we can remove the workflowId part, but the rest are still needed as they invalidate the workflows list page and potentially created tags

Reply:
We are aligned. This was redundant from the beginning.